src/dmd/cond.d

```diff
@@ -118,7 +118,7 @@ extern (C++) final class StaticForeach : RootObject
      */
     bool needExpansion = false;
 
-    extern (D) this(const ref Loc loc,ForeachStatement aggrfe,ForeachRangeStatement rangefe)
+    extern (D) this(const ref Loc loc, ForeachStatement aggrfe, ForeachRangeStatement rangefe)
     {
         assert(!!aggrfe ^ !!rangefe);
```

Comment:
I have a headache trying to understand what this assert tests. Maybe refactor this line also while you're at it?

Reply:
It just tests that only one is true. `!!` is a common way to get the boolean value, and XOR is XOR.
The equivalent using `&&` and `||` wouldn't be much prettier:
```D
assert((aggrfe is null) != (rangefe is null));
```
The above is the shortest expanded version I can think of.

Reply:
It is larger, but it's much clearer IMHO.

Reply:
It all depends on what you're used to. Getting used to shorthand idioms is very useful in the long term.